chain/client/src/client_actor.rs

```diff
@@ -906,6 +908,16 @@ impl ClientActor {
     fn receive_block(&mut self, block: Block, peer_id: PeerId, was_requested: bool) {
         let hash = *block.hash();
         debug!(target: "client", "{:?} Received block {} <- {} at {} from {}, requested: {}", self.client.validator_signer.as_ref().map(|vs| vs.validator_id()), hash, block.header().prev_hash(), block.header().height(), peer_id, was_requested);
+        let head = unwrap_or_return!(self.client.chain.head());
```

Comment:
This breaks the not-completely-unlikely-secneario where
1. The system starts glitching, and a bit of forkfullness appears at the tip (two blocks appear that not everybody knows about)
2. The glitches become more serious, and the nodes generally go down. The skip messages are being sent, but the blocks are not produced.
3. The situation is figured out, and the nodes are restarted. A block with height +500 is produced, and is not accepted by some of the nodes, the system stalls until those nodes restart with full data folder reset.

Is this logic here primarily to avoid computation for the blocks that are produced realtime while we sync (so ~1 block / s)?
If yes, shall we change the condition to "is syncing && !requested"?

Reply:
yeah that sounds good